header/store/store.go

```diff
@@ -36,6 +36,9 @@ var (
 
 // store implements the Store interface for ExtendedHeaders over Datastore.
 type store struct {
+	ctx    context.Context
+	cancel context.CancelFunc
```

Comment:
The context is given as a param in every place where it is now required, no need to change the Store struct.